hazelcast/src/main/java/com/hazelcast/internal/util/DirectoryLock.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hazelcast.internal.util;
+
+import com.hazelcast.core.HazelcastException;
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.util.EmptyStatement;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.nio.channels.ClosedChannelException;
+import java.nio.channels.FileChannel;
+import java.nio.channels.FileLock;
+import java.nio.channels.OverlappingFileLockException;
+
+import static com.hazelcast.nio.IOUtil.closeResource;
+
+/**
+ * A DirectoryLock represents a lock on a specific directory.
+ * <p>
+ * DirectoryLock is acquired by calling {@link #lockForDirectory(File, ILogger)}.
+ */
+public final class DirectoryLock {
```

Comment:
I couldn't see any usage of `DirectoryLock ` inside OSS repo. Is this leftover or for future use?

Reply:
I think initially it was inside the hot-restart code and since now we use it in a more broad context I moved it there. I am OK to move it to another package in EE repo.

Reply:
I am fine with having it in OSS. There's also another similar class `NearCachePreloaderLock`. Later we can remove it and reuse `DirectoryLock ` in nearcache too.